### PR TITLE
support Markdown for warehouse sphinx docs

### PR DESCRIPTION
**Pull request by ewdurbin**
adding support for `.md` files!

Note: This downgrades docutils and doc8 due to a conflict in support for 0.19 in myst-parser being tracked in https://github.com/executablebooks/MyST-Parser/issues/591

### CODE REVIEW

File: requirements/lint.txt [hunk 1]
```diff
@@ -170,7 +171,6 @@ tomli==2.0.1 \
     --hash=sha256:de526c12914f0c550d15924c62d72abc48d6fe7364aa87328337a31007fe8a4f
     # via
     #   black
-    #   doc8
```
Comment (di) [inline requirements/lint.txt:173]: Weird that this went away?

Comment (di) [reply requirements/lint.txt:173]: Oops, got this backwards.

**Review by di — APPROVED** [review]

File: requirements/lint.txt (hunk 1 above)
Comment (ewdurbin) [reply requirements/lint.txt:173]: tomli is a new dependency for doc8 as of 1.0.0

File: requirements/docs.in [hunk 2]
```diff
@@ -1,3 +1,5 @@
 furo
 Sphinx
 sphinxcontrib-httpdomain
+myst-parser
+docutils<0.19
```
Comment (miketheman) [inline requirements/docs.in:5]: Would it make sense to encode the reason in the file itself?

```suggestion
docutils<0.19 # https://github.com/executablebooks/MyST-Parser/issues/591
```

Comment (miketheman) [reply requirements/docs.in:5]: Never mind, it's in the commit message as well, which is good enough for me!

**Pull request merged**